server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -1387,6 +1366,50 @@ private List<ReservedClusterStateHandler<?>> buildReservedStateHandlers(
         return reservedStateHandlers;
     }
 
+    private DiscoveryModule createDiscoveryModule(
+        Settings settings,
+        ThreadPool threadPool,
+        TransportService transportService,
+        NetworkService networkService,
+        ClusterService clusterService,
+        AllocationService allocationService,
+        RerouteService rerouteService,
+        CircuitBreakerService circuitBreakerService,
+        CompatibilityVersions compatibilityVersions,
+        FeatureService featureService
+    ) {
+        GatewayMetaState gatewayMetaState = new GatewayMetaState();
+        FsHealthService fsHealthService = new FsHealthService(settings, clusterService.getClusterSettings(), threadPool, nodeEnvironment);
+
+        DiscoveryModule module = new DiscoveryModule(
+            settings,
+            transportService,
+            client,
+            namedWriteableRegistry,
+            networkService,
+            clusterService.getMasterService(),
+            clusterService.getClusterApplierService(),
+            clusterService.getClusterSettings(),
+            pluginsService.filterPlugins(DiscoveryPlugin.class).toList(),
+            pluginsService.filterPlugins(ClusterCoordinationPlugin.class).toList(),
+            allocationService,
+            environment.configFile(),
+            gatewayMetaState,
+            rerouteService,
+            fsHealthService,
+            circuitBreakerService,
+            compatibilityVersions,
+            featureService
+        );
+
+        modules.add(module, b -> {
+            b.bind(GatewayMetaState.class).toInstance(gatewayMetaState);
```

Comment:
Not sure if these two fit, but wondering if GatewayMetaState and FsHealthService shouldn't belong to DiscoveryModule? At least dependencies here seem to suggest that

Reply:
👍